omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/BedDetailsResource.java

```diff
@@ -51,13 +51,18 @@ public BedDetails getByUniqueId(String id) {
 	}
 	
 	@Override
-	protected void delete(BedDetails bedDetails, String reason, RequestContext requestContext) throws ResponseException {
+	public void delete(String id, String reason, RequestContext requestContext) throws ResponseException {
```

Comment:
Changing to this method means that the controller does not need to load the Bed object (which it doesn't use anyway) and therefore fixes the error.  See my comments on the ticket for further thoughts.